docs/docs/hydro/live-collections/streams.md

```diff
@@ -3,10 +3,119 @@ sidebar_position: 2
 ---
 
 # Streams
-You can view the full API documentation for Streams [here](pathname:///rustdoc/hydro_lang/stream/struct.Stream).
+Streams are the most common type of live collection in Hydro; they can be used to model streaming data collections, a feed of API requests, or even time-based intervals. A `Stream` represents a sequence of elements, with new elements being asynchronously appended to the end of the sequence. Streams can be transformed using APIs like `map` and `filter`, based on Rust [iterators](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html). You can view the full API documentation for Streams [here](pathname:///rustdoc/hydro_lang/stream/struct.Stream).
```

Comment:
I guess it's not obvious to me how a time-based interval is a `Stream`. Given that it sounds kind of esoteric, maybe postpone mentioning it to a later FAQ or the like that talks about "which do I use? `Stream`, `Singleton` or `Optional`?"